components/comments/commentForm.js

```diff
@@ -1,94 +1,104 @@
 import { useState } from "react";
-import { addComment, addCommentator, updateComment } from "../../lib/comment";
 import { faMessage } from "@fortawesome/free-solid-svg-icons";
 import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import axios from "axios";
+import config from "../../config";
+const STRAPI_URL = config.STRAPI_URL;
 
-export default function CommentForm({ post }) {
-  const [postId, commentId] = post;
-
+export default function CommentForm({ post, onNewComment }) {
+  var [name, setName] = useState("");
+  var [email, setEmail] = useState("");
+  var [message, setMessage] = useState("");
   let [showComment, setShowResults] = useState(false);
   const onClick = () => setShowResults(true);
+  const [postId, parentId] = post;
 
-  const [title, setTitle] = useState("");
-  const [email, setEmail] = useState("");
-  const [comment, setComment] = useState("");
+  async function handleSubmit(event) {
+    event.preventDefault();
 
-  async function handleSubmit() {
-    const commentData = {
-      data: {
-        comment: comment,
-        post: postId,
-        parentId: commentId ? commentId : null,
-      },
+    const userData = {
+      username: name,
+      email: email,
     };
 
-    const [status, comments] = await addComment(commentData);
-    if (status === 200) {
-      const userData = {
+    const response = await axios.post(
+      `${STRAPI_URL}/v1/comments/createOrUpdateUser`,
+      {
+        userData,
+      }
+    );
+    const userId = response.data.userId;
+
+    if (response.status === 200) {
+      const commentData = {
         data: {
-          username: title,
-          email: email,
-          comments: comments.data.id,
+          comment: message,
+          post: postId,
+          parent_id: parentId ? parentId : null,
+          user_id: userId,
```

Comment:
No need to pass userId from here and even call the `createOrUpdateUser` API. What you can do is,
- Prepare commentData without user_id, and pass username and email with this data only
- Call API which inserts a comment.
- At strapi, you can first create the user and then get userID and then create the comment an return whole object here